src/glue/jobs/s3_to_json.py

```diff
@@ -16,6 +16,237 @@
 import boto3
 from awsglue.utils import getResolvedOptions
 
+DATA_TYPES_WITH_SUBTYPE = ["HealthKitV2Samples", "HealthKitV2Statistics"]
+
+def transform_object_to_array_of_objects(
+        json_obj_to_replace: dict,
+        key_name: str,
+        key_type: type,
+        value_name: str,
+        value_type: type,) -> list:
+    """
+    Transforms a dictionary object into an array of dictionaries with specified
+    key and value types.
+
+    This function takes a dictionary object `json_obj_to_replace` and transforms
+    it into an array of dictionaries, where each dictionary in the array contains
+    two key-value pairs. The keys are specified by the `key_name` and `value_name`
+    parameters, and the values are converted to the corresponding types specified
+    by the `key_type` and `value_type` parameters.
+
+    Parameters:
+        json_obj_to_replace (dict): The input dictionary object to be
+            transformed into an array of dictionaries.
+        key_name (str): The name of the key in the output dictionaries.
+        key_type (type): The type to which the values corresponding to `key_name`
+            should be converted in the output dictionaries.
+        value_name (str): The name of the value in the output dictionaries.
+        value_type (type): The type to which the values corresponding to `value_name`
+            should be converted in the output dictionaries.
+
+    Returns:
+        list: An array of dictionaries, where each dictionary contains
+            two key-value pairs:
+              - The key specified by `key_name` with the value converted to the type specified by
+                `key_type`.
+              - The value specified by `value_name` with the value converted to the type specified
+                by `value_type`.
+
+    Examples:
+        json_obj = {'TimeOffsetHeartRateSamples': {"0": 62, "1": 63, "2": 62}}
+        transformed_array = transform_object_to_array_of_objects(
+            json_obj,
+            key_name='OffsetInSeconds',
+            key_type=int,
+            value_name='HeartRate',
+            value_type=int
+        )
+
+        # Resulting `transformed_array`:
+        # [
+        #     {'OffsetInSeconds': 0, 'HeartRate': 62},
+        #     {'OffsetInSeconds': 1, 'HeartRate': 63},
+        #     {'OffsetInSeconds': 2, 'HeartRate': 62},
+        # ]
+    """
+    array_of_obj = [
+            {
+                key_name: key_type(k),
```

Comment:
Will there ever be cases where you won't be able to cast the value?

Things like (int('')) might cause an error.

Reply:
I don't know. This data comes from Garmin (I believe CareEvolution is just an intermediary, they don't touch this part of the data), so it depends on what processes Garmin has in place so that they aren't writing incorrect types. I didn't experience any issues with the pilot data.